sdk/trace/span.go

```diff
@@ -347,17 +347,24 @@ func (s *recordingSpan) End(options ...trace.SpanEndOption) {
 		return
 	}
 
+	config := trace.NewSpanEndConfig(options...)
+
 	// Store the end time as soon as possible to avoid artificially increasing
 	// the span's duration in case some operation below takes a while.
-	et := internal.MonotonicEndTime(s.startTime)
+	var et time.Time
+	// Setting endTime to non-zero marks the span as ended and not recording.
+	if config.Timestamp().IsZero() {
+		et = internal.MonotonicEndTime(s.startTime)
+	} else {
+		et = config.Timestamp()
+	}
 
 	// Do relative expensive check now that we have an end time and see if we
 	// need to do any more processing.
 	if !s.IsRecording() {
```

Comment:
This method call requires `endTime` to be set (per the comment), so we can't move setting endTime after it.
You can see it being used here: https://github.com/rueian/opentelemetry-go/blob/c6b2456ccd1622ffd3f2b4988e0a34788e8ad576/sdk/trace/span.go#L179

Reply:
Hi @damemi, thank you for reviewing the code. However, I am not sure what you mean by that.

This PR doesn't change the timing of setting the `s.endTime` relative to `s.IsRecording()`.

Reply:
Urgh you're right, I misread the diff. Sorry.